astropy/io/ascii/cds.py

```diff
@@ -164,14 +208,302 @@ def get_cols(self, lines):
+        # maxEnt: maximum number of digits places before decimal point.
+        maxSize, maxprec, maxDec, maxEnt = 1, 0, 0, 1
+        sign = False
+        fformat = 'F'
+        fmt = [0, 0, 0, 0, False]
+
+        # find max sized value in the col
+        for rec in col:
+            # skip null values
+            if rec is None:
+                continue
+            s = str(rec)
+
+            if self.__splitFloatFormat(s, fmt) is True:
+                if fformat == 'F':
+                    maxSize, maxprec, maxDec = 1, 0, 0
+                # scientific notation
+                fformat = 'E'
+            else:
+                if fformat == 'E': continue
+
+            if maxprec < fmt[1]: maxprec = fmt[1]
+            if maxDec < fmt[3]: maxDec = fmt[3]
+            if maxEnt < fmt[2]: maxEnt = fmt[2]
+            if maxSize < fmt[0]: maxSize = fmt[0]
+            if fmt[4]: sign = True
+
+        if fformat == 'E':
+            col.meta.size = maxSize
+            if sign: col.meta.size += 1
+            col.fortran_format = fformat + str(col.meta.size) + "." + str(maxprec)
+            col.format = str(col.meta.size) + "." + str(maxDec) + "e"
+        else:
+            col.meta.size = maxEnt + maxDec + 1
+            if sign: col.meta.size += 1
+            col.fortran_format = fformat + str(col.meta.size) + "." + str(maxDec)
+            col.format = col.fortran_format[1:] + "f"
+
+    def writeByteByByte(self):
+        """
+        Writes byte-by-byte description of the table.
+        :param table: `astropy.table.Table` object.
+        """
+        # get column widths.
+        vals_list = []
+        col_str_iters = self.data.str_vals()
+        for vals in zip(*col_str_iters):
+            vals_list.append(vals)
+
+        for i, col in enumerate(self.cols):
+            col.width = max([len(vals[i]) for vals in vals_list])
+            if self.start_line is not None:
+                col.width = max(col.width, len(col.info.name))
+        widths = [col.width for col in self.cols]
+
+        startb = 1
+        # set default width of Start Byte, End Byte, Format and
+        # Label columns in the ByteByByte table.
+        sz = [0, 0, 1, 7]
+        l = len(str(sum(widths)))
+        if l > sz[0]:
+            sz[0] = l
+            sz[1] = l
+        for column in self.cols:
+            if len(column.name) > sz[3]:
+                sz[3] = len(column.name)
+
+        buff = ""
+        maxDescripSize = 16
+        nsplit = sz[0] + sz[1] + sz[2] + sz[3] + 16
+        # format string for Start Byte and End Byte
+        fmtb = "{0:" + str(sz[0]) + "d}-{1:" + str(sz[1]) + "d} {2:" + str(sz[2]) + "s}"
+
+        bbb = Table(names=['Bytes', 'Format', 'Units', 'Label', 'Explanations'],
+                    dtype=[str]*5)
+
+        for i, col in enumerate(self.cols):
+            # check if column is MaskedColumn
+            if isinstance(col, MaskedColumn):
```

Comment:
```suggestion
            col.hasNull =  isinstance(col, MaskedColumn)
```
And then delete the three lines below this. The fact that this test is only one line long also means that we don't need the `hasNull` attribute at all. Instead, we can go through the code and replace every occurrence of
`if col.hasNull` with `if isinstance(col, MaskedColumn)`. Not much longer, but save us an extra attribute. 
[At least, that's true until we have cases with `hasNull` that are not instances of MaskedColumn. Not sure aobut mixin columns with e.g. coordinates. So maybe it's better to leave `hasNull` for now, but the check to set True or False can still be one line only.

Reply:
Okay. I am saving `col.hasNull` for the time being.